modules/queue/manager.go

```diff
@@ -427,6 +429,14 @@ func (q *ManagedQueue) SetPoolSettings(maxNumberOfWorkers, boostWorkers int, tim
 	}
 }
 
+// NumberInQueue returns the number of items in the queue
+func (q *ManagedQueue) NumberInQueue() int64 {
+	if pool, ok := q.Managed.(ManagedPool); ok {
+		return pool.NumberInQueue()
+	}
+	return -1
```

Comment:
I think we should add a error log here and return 0

Later: `q.byteFIFO.Len(q.terminateCtx) + q.WorkerPool.NumberInQueue()`, if we return -1 here, that expression is strange .....

Reply:
Not everything is a managed pool.

Your quoted expression is not related to this.

Reply:
I know it not really affected, but related. Using `-1` to indicate a non-sense call would make it difficult to debug root problems in future, we can not guarantee that nobody writes incorrect code. The Golang best practice for this case is returning `(int, error)`

Reply:
These are UI related functions. If they returned an err it would make the templates extremely complex. That is why they don't return an error.